### PR TITLE
fix: Prevent the Playground from crashing on a form error.

### LINKED ISSUE #3132: Playground crashes on `{"type":["string"]}`
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

core

### Version

Not sure, whatever version the playground uses

### Current Behavior

The playground goes blank when using `{"type":["string"]}` as the schema.

### Expected Behavior

The playground continues to work.

### Steps To Reproduce

1. Open the playground.
2. Paste `{"type":["string"]}` as the JSON schema.
3. Page goes blank and console shows these errors.
```
Error: No widget 'text' for type 'string'
    at ce (main.js:1:12375)
    at StringField (main.js:1:33777)
    at ss (740.js:75:3259258)
    at $a (740.js:75:3311307)
    at Nl (740.js:75:3298581)
    at kl (740.js:75:3298509)
    at xl (740.js:75:3298372)
    at ml (740.js:75:3295359)
    at 740.js:75:3245180
    at t.unstable_runWithPriority (740.js:75:3342116)
        
Uncaught Error: Model is disposed!
Error: Model is disposed!
    at e._assertNotDisposed (740.js:75:1600124)
    at e.validateRange (740.js:75:1609581)
    at hi._setState (740.js:75:1137600)
    at hi.setState (740.js:75:1136990)
    at pi.setStates (740.js:75:1140124)
    at pi.setSelections (740.js:75:1139986)
    at Hi._interpretCommandResult (740.js:75:1157149)
    at Hi._executeEditOperation (740.js:75:1156658)
    at 740.js:75:1160599
    at Hi._executeEdit (740.js:75:1158890)
    at 740.js:75:710111
        
Uncaught (in promise) Error: Model is disposed!
    at e._assertNotDisposed (740.js:75:1600124)
    at e.getVersionId (740.js:75:1604381)
    at e._deltaDecorationsImpl (740.js:75:1622267)
    at e._setTrackedRange (740.js:75:1618421)
    at hi._updateTrackedRange (740.js:75:1136517)
    at hi.startTrackingSelection (740.js:75:1136341)
    at pi.startTrackingSelections (740.js:75:1139111)
    at Hi._executeEdit (740.js:75:1158949)
    at Hi.paste (740.js:75:1160575)
    at 740.js:75:1220848
```

### Environment

```markdown
- OS: Windows 10
- Browser: Edge 105.0.1343.42
```

### Anything else?

Potentially related to #2179

### PR DESCRIPTION
**Pull request by nickgros**
Fixes #3132, #3095, #2787, #2277, #2179, #2163, #1762, and #1002;

Potentially fixes other issues that I couldn't find.

image

### CODE REVIEW

File: packages/playground/src/ErrorBoundary.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  /** Update state so the next render will show the fallback UI. */
+  static getDerivedStateFromError(error) {
+    return { hasError: true, error: error };
+  }
+
+  resetErrorBoundary() {
+    this.setState({ hasError: false, error: null });
+  }
```
Comment (heath-freenome) [inline packages/playground/src/ErrorBoundary.jsx:17]: If you turn this into a fat-arrow function then you don't need the bind above


```suggestion
  resetErrorBoundary = () => {
    this.setState({ hasError: false, error: null });
  }
```

File: packages/playground/src/app.jsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import Form, { withTheme } from "@rjsf/core";
 import { shouldRender } from "@rjsf/utils";
 import DemoFrame from "./DemoFrame";
 import localValidator from "@rjsf/validator-ajv6";
```
Comment (heath-freenome) [inline packages/playground/src/app.jsx:8]: Reorder this so all external imports are first
```suggestion
import localValidator from "@rjsf/validator-ajv6";

import DemoFrame from "./DemoFrame";
```

**Review by heath-freenome — CHANGES_REQUESTED** [review]

**Comment by heath-freenome** [comment]
@nickgros Have you verified that this will handle all the uses cases you've indicated will be fixed?

File: packages/playground/src/ErrorBoundary.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false, error: null };
+    this.resetError = this.resetErrorBoundary.bind(this);
```
Comment (nickgros) [inline packages/playground/src/ErrorBoundary.jsx:7]:
```suggestion
```

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**